dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/HibernateTrackedEntityInstanceStoreTest.java

```diff
@@ -0,0 +1,162 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker;
+
+import static org.hisp.dhis.tracker.Assertions.assertNoErrors;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.hisp.dhis.analytics.AggregationType;
+import org.hisp.dhis.common.QueryItem;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.trackedentity.TrackedEntityInstanceQueryParams;
+import org.hisp.dhis.trackedentity.hibernate.HibernateTrackedEntityInstanceStore;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.webapi.controller.event.mapper.OrderParam;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class HibernateTrackedEntityInstanceStoreTest extends TrackerTest
+{
+
+    @Autowired
+    private TrackerImportService trackerImportService;
+
+    @Autowired
+    private HibernateTrackedEntityInstanceStore teiStore;
+
+    private TrackedEntityInstanceQueryParams params;
+
+    @Override
+    protected void initTest()
+        throws IOException
+    {
+        setUpMetadata( "tracker/simple_metadata.json" );
+        User userA = userService.getUser( "M5zQapPyTZI" );
+        assertNoErrors(
+            trackerImportService
+                .importTracker( fromJson( "tracker/tracked_entity_basic_data_for_ordering.json", userA.getUid() ) ) );
+        manager.flush();
+
+        params = new TrackedEntityInstanceQueryParams();
+    }
+
+    @ParameterizedTest
+    @MethodSource( "provideOrderParamsAndExpectedResult" )
+    void whenOrderingByBasicStaticFieldsThenEntitiesAreSortedAccordingly( String input, String expected )
+    {
+        params.setOrders( List.of( new OrderParam( input, OrderParam.SortDirection.ASC ) ) );
+
+        List<Long> list = teiStore.getTrackedEntityInstanceIds( params );
+        list = reduceListValuesToStartCountingFromOne( list );
```

Comment:
If you are doing these two in every test you might as well create a private wrapper function `getTrackedEntityInstanceIds` so this just becomes one call like `List<Long> list = getTrackedEntityInstanceIds( params );`